### PR TITLE
detect: Propagate app-layer decode events 

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #6624 

This PR propagates app-layer decode events so anomaly logging is possible.

Without this PR, file compression events and exceeding the inspection buffer limit are recorded but never exposed.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [4482](https://redmine.openinfosecfoundation.org/issues/4482)

Updates
- Increased scope of `det_ctx` packet pointer to cover full detect operation. Previously, scope was limited to a transaction.

Describe changes:
- Move un-exposed events to app-layer-events 
- Propagate app-layer decode events to packet
- Eliminates detection context event as events are now added to app layer event element of packet.

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/app-layer-events.c [hunk 1]
```diff
@@ -135,6 +143,7 @@ void AppLayerDecoderEventsSetEvent(Flow *f, uint8_t event)
     AppLayerDecoderEvents *new = events;
     AppLayerDecoderEventsSetEventRaw(&events, event);
     if (events != new)
+        SCLogNotice("Setting events to %p", events);
```
Comment (jasonish) [inline src/app-layer-events.c:146]: Probably shoudln't be notice.

Comment (jlucovsky) [reply src/app-layer-events.c:146]: You're correct -- I'll remove this line. I don't think it's useful for debugging either.

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on tlpw2_single_alerts_cmp.
 | field | test | baseline | % |
|--|--|--|--| 
| | generic_stats_chk |
| .flow.spare | 1991488 | 1866612 | 106.69% | 

Pipeline 5031

**Comment by jlucovsky** [comment]
Continued in #6643

**Pull request closed without merging**